include/uart.h

```diff
@@ -363,30 +370,44 @@ static inline void uart_irq_rx_disable(struct device *dev)
 }
 
 /**
- * @brief Check if nothing remains to be transmitted
+ * @brief Check if UART TX block finished transmission
+ *
+ * @details Check if any outgoing data buffered in UART TX block was
+ * fully transmitted and TX block is idle. When this condition is
+ * true, UART device (or whole system) can be power off. Note that
+ * this function is *not* useful to check if UART TX can accept more
+ * data, use uart_irq_tx_ready() for that. This function must be called
+ * in a UART interrupt handler, or its result is undefined. Before
+ * calling this function in the interrupt handler, uart_irq_update()
+ * must be called once per the handler invocation.
  *
  * @param dev UART device structure.
  *
  * @retval 1 If nothing remains to be transmitted.
  * @retval 0 Otherwise.
  */
-static inline int uart_irq_tx_empty(struct device *dev)
+static inline int uart_irq_tx_complete(struct device *dev)
 {
 	const struct uart_driver_api *api = dev->driver_api;
 
-	if (api->irq_tx_empty) {
-		return api->irq_tx_empty(dev);
+	if (api->irq_tx_complete) {
+		return api->irq_tx_complete(dev);
 	}
 
 	return 0;
 }
```

Comment:
you need to add attribute __deprecated, doxygen @deprecated is not enough :)
Look at include/dma.h for example.